react-native/react/native/pinentry.wrapper.desktop.js

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from '../base-react'
+import ReactDOM from 'react-dom'
+import Pinentry from '../pinentry'
+import ipc from 'ipc'
+
+class PinentryWrapper extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {}
+    ipc.on('pinentryGotProps', props => {
```

Comment:
I agree, I think it would be cool to have the component be given dispatch actions that look like our normal dispatch actions, but use ipc.send under the hood.

If we could also abstract out how props are being passed in, I don't think we'd need anything else (like a reducer). All the redux logic would stay in the main app, and it would treat the popup as just another component!

## Quick sketch
I think we could build something that looks like this: (very rough)

```js
<RemoteWindow
  component={'react-native/react/pinentry.wrapper'} // this is what the require would call
  pinEntryProp1={'some reason for the pinEntry'} />
```

pinentry.wrapper has these Props

```js
PinentryWrapperProps = {
  pinEntryProp1: string,
  onSubmit: (pass: string) => void
}
```

and we make our own `connect` that does the dispatch/ipc stuff. and it should do the logic of getting the props from ipc.

```js
// inside pinentry.wrapper.js
import {onSubmit} from './actions/pinentry'
... 
export default remoteConnect(
  state => {return {}},
  dispatch => { // This dispatch is our own dispatch that actually calls ipc.send
    {onSubmit: (pass) => dispatch(onSubmit(pass))}
  }
)(PinentryWrapper)
```

The remote window component receives ipc.sends, and dispatches them into our existing redux stuff.

```js
// inside remote-window.js
...
componentWillMount () {
  const windowId = ...
  ipc.on('actionDispatch', (event, arg) => {
    if (event.sender === windowId) {
     this.props.dispatch(arg);
    }
  })
}
// Somewhere else here we send the props and open the browser window
// something like:
window.loadUrl(`file://${__dirname}/remoteWindow.html#${componentWrapperPath}`)
// so the remote window html requires the component wrapper path and uses that to render
```